### PR TITLE
mes-3563 - implement set isolation level

### PR DESCRIPTION
**Pull request by roberthall12**
# Description and relevant Jira numbers

Story: https://jira.dvsacloud.uk/browse/MES-2621
Subtask: https://jira.dvsacloud.uk/browse/MES-3563

- Adding serialisable isolation level to the retry transactions
- Added test for docker db to ensure isolation level is changed.

## Pull Request checklist

- [x] [WIP] tag removed from PR title
- [x] PR has an understandable description

## Git feature branch checklist

- [x] branch name comply with our branching strategy
- [x] git branch contains relevant JIRA ticket number
- [x] branch rebased against the latest develop

## Sign off process checklist

- [x] Code has been tested manually
- [ ] Tested by QA
- [ ] PO's approval
- [x] Reviewers selected in Github
- [x] PR link added to JIRA

### CODE REVIEW

File: src/functions/retryProcessing/application/RetryProcessor.ts [hunk 1]
```diff
@@ -25,6 +26,7 @@ export class RetryProcessor implements IRetryProcessor {
 
   async processSuccessful(): Promise<number> {
     try {
+      await this.connection.promise().query(setIsolationLevelSerializable);
```
Comment (rstepnowski) [inline src/functions/retryProcessing/application/RetryProcessor.ts]: Transaction isolation level is a Session variable - once set, it applies to all following transactions on the same connection - so you just need to call it once at the beginning of the processing (probably where the connection is being created in the first place)

**Review by rossfauldsbjss — APPROVED** [review]
Looks good.

**Review by szabikr — APPROVED** [review]
After the comment raised @rstepnowski is sorted I'm happy with it.

File: src/functions/retryProcessing/application/RetryProcessor.ts (hunk 1 above)
Comment (roberthall12) [reply src/functions/retryProcessing/application/RetryProcessor.ts]: Have moved the query to where the connection is being established to remove sending duplicate queries.

**Pull request merged**